API.Tests/ControllersTests/TrustsControllerTests.cs

```diff
@@ -0,0 +1,616 @@
+                                                        errorHandlerMock.Object);
+
+
+            //Execute
+            var result = await trustsController.GetById(trustId);
+            var castedResult = (ObjectResult)result.Result;
+
+            //Assert
+
+            //Error handler should not be called when repo returns valid result
+            errorHandlerMock.Verify(h => h.LogAndCreateResponse(It.IsAny<RepositoryResultBase>()), Times.Never);
+            //Mapper not to be called when repo returns null but valid result
+            mapperMock.Verify(m => m.Map(It.IsAny<GetTrustsD365Model>()), Times.Never);
+
+            //Final result should be a 404 - Not Found with the correct message
+            Assert.Equal("The trust with the id: 'a16e9020-9123-4420-8055-851d1b672fa9' was not found", castedResult.Value);
+            Assert.Equal(404, castedResult.StatusCode);
+        }
+
+        [Fact]
+        public async void GetById_TrustFound_Returns200OkResult()
+        {
+            //Arrange
+            var trustId = Guid.Parse("a16e9020-9123-4420-8055-851d1b672fa9");
+
+            //Set up mock repository to return a valid and set result
+            var trustsRepoMock = new Mock<ITrustsRepository>(); 
+            trustsRepoMock.Setup(m => m.GetTrustById(trustId))
+                          .ReturnsAsync(new RepositoryResult<GetTrustsD365Model>
+                          {
+                              Result = new GetTrustsD365Model
+                              {
+                                  Id = trustId
+                              }
+                          });
+
+            var mapperMock = new Mock<IMapper<GetTrustsD365Model, GetTrustsModel>>();
+
+            //Set up mapper to return a slim result when called with the expected input
+            mapperMock.Setup(m => m.Map(It.Is<GetTrustsD365Model>(p => p.Id == trustId)))
+                      .Returns(new GetTrustsModel
+                      {
+                          Id = trustId,
+                          Address = "Some address"
+                      });
+
+            var errorHandlerMock = new Mock<IRepositoryErrorResultHandler>();
+
+            var trustsController = new TrustsController(trustsRepoMock.Object,
+                                                        _academiesRepository.Object,
+                                                        mapperMock.Object,
+                                                        _getAcademiesMapper.Object,
+                                                        errorHandlerMock.Object);
+
+
+            //Execute
+            var result = await trustsController.GetById(trustId);
+            var castedResult = (ObjectResult)result.Result;
+
+            //Assert
+
+            //Error handler should not be called when repo returns valid result
+            errorHandlerMock.Verify(h => h.LogAndCreateResponse(It.IsAny<RepositoryResultBase>()), Times.Never);
```

Comment:
Suggestion: I think these tests are really, really useful but I'd be tempted to break them out into separate test methods.  Reason being that if the first one fails, you won't know about the others until you fix the bug and run again.  I think it gives a clearer indication as to the level of regression in a CI build if there were, for example, 5 test failures, rather than just the first one it encountered in a single routine.  I'm less worried about tests which just check property values but the checks on mapping and error responses being generated I feel would be useful as standalone tests.  Appreciate this is a fair amount of work with little to gain and not much time left, so feel free to ignore as appropriate.

Reply:
"Appreciate this is a fair amount of work with little to gain " - Fully agree on that one. I agree in principle but it's hardly maximizing the gains given the effort.